File: tests/test_haskell/Services/OrderService.hs
module Services.OrderService
    ( processOrder
    , findAllOrders
    ) where

import Models.Order
import Models.User

processOrder :: Order -> IO ()
processOrder order = putStrLn $ "Processing order: " ++ show (orderId order)

findAllOrders :: IO [Order]
findAllOrders = return []

File: tests/test_lua/services/order_service.lua
local Order = require("models.order")
local User = require("models.user")

local OrderService = {}
OrderService.__index = OrderService

function OrderService.new()
    local self = setmetatable({}, OrderService)
    self.orders = {}
    return self
end

function OrderService:process(order)
    table.insert(self.orders, order)
    print("Processing order: " .. order.id)
end

return OrderService
